File: R_scripts/script_3.R
#!/usr/bin/env Rscript
#
# R_scripts/script_3.R
# --------------------------------------------------------
# This script is a working file for this tutorial. You
# will need to edit this file to implement reviewer
# comments.
#
# You can run this script "individually" by 
# calling the `test_M` function defined below.
#

library(ggplot2)

draw_M <- function(canvas) {
  # Adds a letter "M" to an existing ggplot2 context

  # read in data
  M <- read.csv(file = 'data/letter_3.csv',
                header = FALSE)
  names(M) <- c('x', 'y')
  M[nrow(M) + 1,] <- M[1,]

  # plot data
  canvas +
    geom_polygon(
      data=M,
      fill='black',
      color='black',
      size=1,
      aes(x=x, y=y))
}


test_M <- function() {
  # Displays just the M body
  
  canvas <- ggplot(NULL) + 
    theme(axis.title=element_blank(),
          axis.text=element_blank(),
          axis.ticks=element_blank(),
          panel.background=element_blank(),
          panel.grid=element_blank())
  
  draw_M(canvas)  
}

## Solution ----
# draw_M <- function(canvas) {
#   # Adds a M body to an existing ggplot2 context
# 
#   # read in data
#   M <- read.csv(file = 'data/letter_3.csv',
#                     header = FALSE)
#   names(M) <- c('x', 'y')
#   M[nrow(M) + 1,] <- M[1,]
# 
#   # plot data
#   canvas +
#     geom_polygon(
#       data=M,
#       fill=rgb(1, 0.8, 0),  # per reviewer comment
#       color='black',
#       size=1,
#       aes(x=x, y=y))
# }

File: python_scripts/script_3.py
#!/usr/bin/env python
'''
	python_scripts/script_3.py
	--------------------------------------------------------
	This script is a working file for this tutorial. You
	will need to edit this file to implement reviewer
	comments.

	You may run this script individually via

		$ python python_scripts/script_3.py
		
'''


def drawM():
	'''Draw the M body'''
	# read in data
	M = pd.read_csv('data/letter_3.csv',
					header = None)
	x, y = list(M[0]), list(M[1])
	x.append(x[0])
	y.append(y[0])

	# plot data
	ax.fill(x, y, color = 'black')
	ax.plot(x, y, color = 'black', linewidth = 2)

## Solution ----
# def drawM():
# 	'''Draw the M body'''
# 	# read in data
# 	M = pd.read_csv('data/letter_3.csv',
# 					header = None)
# 	x, y = list(M[0]), list(M[1])
# 	x.append(x[0])
# 	y.append(y[0])
#
# 	# plot data
# 	ax.fill(x, y, color = (1, 0.8, 0))  # per reviewer comment
# 	ax.plot(x, y, color = 'black', linewidth = 2)


# You will not need to edit the code below
if __name__ == '__main__':
	# We are running this script on its own
	import pandas as pd
	import matplotlib.pyplot as plt
	fig, ax = plt.subplots(1, 1)		
	ax.axis('off')
	drawM()
	plt.show()
else:
	# We are running this script from within main_script.py
	# Thus all the dependencies and globals have already 
	# been imported.
	drawM()